substrate/frame/revive/src/evm/runtime.rs

```diff
@@ -333,12 +333,27 @@ pub trait EthExtra {
 			})?;
 
 		let call = if let Some(dest) = to {
-			crate::Call::call::<Self::Config> {
-				dest,
-				value,
-				gas_limit,
-				storage_deposit_limit,
-				data,
+			if dest == RUNTIME_PALLETS_ADDR {
+				let call = CallOf::<Self::Config>::decode(&mut &data[..]).map_err(|_| {
```

Comment:
Needs a decode_with_max_depth or something. You need to check where it is decoded elsewhere.

Reply:
probably fine to apply the same limit as in https://github.com/paritytech/polkadot-sdk/blob/pg/fix-extension-weight/substrate/frame/executive/src/lib.rs?plain=1#L801-L798

https://github.com/paritytech/polkadot-sdk/pull/8901/commits/eadfed36232e721069eebb4563a47c6106e457e8